webapp/src/js/services/enketo.js

```diff
@@ -574,13 +574,29 @@ angular.module('inboxServices').service('Enketo',
           return xmlToDocs(doc, form.getDataStr({ irrelevant: false }));
         })
         .then(function(docs) {
-          if (geolocation) {
-            // Only update geolocation if one is provided.  Otherwise, maintain
-            // whatever is already set in the docs.
-            docs.forEach(function(doc) {
-              doc.geolocation = geolocation;
-            });
+          if (geoHandle) {
+            return geoHandle()
```

Comment:
This will block until the promise resolves or rejects, right? So it may be up to 30 seconds with the form submission spinning? I guess that's probably ok?

Reply:
Yeah. So there is a question on the ticket about what we want that to be. We can make that any number we like, it can be 10s, or 5 etc, but unlike current master the timeout will actually matter.

Reply:
30 seconds is probably fine. I mean, most forms will take longer than that to complete...